polygeist/include/mlir/Dialect/Polygeist/Utils/TransformUtils.h

```diff
@@ -230,6 +231,14 @@ class LoopTools {
                    const LoopVersionCondition &versionCond) const;
 };
 
+//===----------------------------------------------------------------------===//
+// SYCL related functions
+//===----------------------------------------------------------------------===//
+
+/// Return true if the given function is potentially a SYCL kernel body
+/// function.
+bool isPotentialKernelBodyFunc(FunctionOpInterface);
```

Comment:
I would drop "Potential" from the name. It is either the function called from gpu.func or not.

Reply:
but it also checks for users to be only tail calls.

Reply:
updated function description, please check if it makes more sense. eventually this utility should be removed, and we can use the marker added by sema.